exercises/lens-person/examples/success-standard/src/Person.hs

```diff
@@ -36,17 +37,44 @@ data Gregorian = Gregorian {
       _day   :: Int
     }
 
-makeLenses ''Person
-makeLenses ''Name
-makeLenses ''Born
-makeLenses ''Address
-makeLenses ''Gregorian
+type Lens s t a b = forall f. Functor f => (a -> f b) -> s -> f t
+type Lens' s a = Lens s s a a
+type Getting s a = (a -> Const a a) -> s -> Const a s
+type Setting s t a b = (a -> Identity b) -> s -> Identity t
 
-gregorianDay :: Iso' Gregorian Day
-gregorianDay = iso toDay fromDay
-  where
-    toDay (Gregorian y m d) = fromGregorian y m d
-    fromDay d' = let (y, m, d) = toGregorian d' in Gregorian y m d
+lens :: (s -> a) -> (s -> b -> t) -> Lens s t a b
+lens sa sbt afb s = fmap (sbt s) (afb (sa s))
```

Comment:
these names are completely useless, but I'm not sure I can do better. I suppose `sa` can be named `get` and `sbt` can be `set`, but I'm not sure about `afb`